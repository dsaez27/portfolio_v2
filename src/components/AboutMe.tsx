import { path } from "@/helpers";

const AboutMe = () => {
    return (
        <section id="about" className="h-full w-full">
            <h2 className="after-content-[''] text-3xl font-semibold text-primary after:mt-1  after:block after:h-1 after:w-20 after:bg-primary">
                Sobre mí
            </h2>
            <div className="mt-20 mb-10 flex flex-col items-center justify-center">
                <img
                    src={path("avatar.webp", "images")}
                    alt="Profile"
                    className="-mt-12 h-32 w-32 rounded-full"
                />
                <div className="my-5 flex flex-col items-center justify-center gap-1">
                    <h1 className="text-4xl font-semibold text-white">
                        Daniel Sáez
                    </h1>
                    <span className="text-sm text-white">
                        Front End Developer
                    </span>
                    <p className="mb-4 w-3/4 text-center leading-relaxed text-gray-300 max-sm:w-full">
                        ¡Hola! Soy un programador con conocimientos en React,
                        Bootstrap, Tailwind, CSS3, Redux, Sass, Git, Github, JS
                        y TS. En mi sitio encontrarás mis proyectos personales y
                        habilidades de programación. Si necesitas ayuda con el
                        desarrollo de un proyecto, contáctame. ¡Gracias por
                        visitar mi sitio web!
                    </p>
                </div>
            </div>
        </section>
    );
};

export default AboutMe;
