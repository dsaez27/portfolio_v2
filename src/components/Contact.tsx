import { useEffect, useRef, useState } from "react";
import { useForm } from "react-hook-form";
import * as Yup from "yup";
import { useMutation } from "react-query";
import { yupResolver } from "@hookform/resolvers/yup";
import ReCAPTCHA from "react-google-recaptcha";
import axios from "axios";

interface FormValues {
    name: string;
    email: string;
    message: string;
    reCaptchaValue: string | null;
}

const ContactSchema = Yup.object().shape({
    name: Yup.string().required("Nombre es requerido"),
    email: Yup.string().email("Invalid email").required("Email es requerido"),
    message: Yup.string().required("Mensaje es requerido"),
});

const site_key: string = "6LeAG4slAAAAAN0ER59FUnK102kZi3hcn-9sDHtC";

const Contact = () => {
    const [serverError, setServerError] = useState<string | null>(null);
    const [success, setSuccess] = useState<boolean | null>(null);
    const [reCaptchaValue, setReCaptchaValue] = useState<string | null>(null);

    const {
        register,
        handleSubmit,
        formState: { errors },
        reset,
    } = useForm<FormValues>({
        resolver: yupResolver(ContactSchema),
    });

    const reCaptchaRef = useRef<ReCAPTCHA>(null);

    const { mutate, status } = useMutation(
        async (data: FormValues) => {
            const response = await axios.post(
                "http://localhost:5001/submit-form",
                {
                    ...data,
                    reCaptchaValue,
                }
            );

            if (!response.data.success) {
                throw new Error("Something went wrong");
            }
        },
        {
            onSuccess: () => {
                setServerError(null);
                setSuccess(true);
                reset();
                reCaptchaRef.current?.reset();
            },
            onError: (error: Error) => {
                setServerError(error.message);
                setSuccess(false);
            },
        }
    );

    const onChange = (value: string | null) => {
        setReCaptchaValue(value);
    };

    const onSubmit = (data: FormValues) => {
        if (reCaptchaValue) {
            mutate(data);
        } else {
            setServerError("Por favor verifica que no eres un robot");
        }
    };

    return (
        <section id="contact" className="h-full w-full">
            <h2 className="after-content-[''] text-3xl font-semibold text-primary after:mt-1  after:block after:h-1 after:w-20 after:bg-primary">
                Contacto
            </h2>
            <form
                onSubmit={handleSubmit((data) => onSubmit(data))}
                className="my-10 flex w-1/2 flex-col items-start gap-2 max-md:w-full"
            >
                <input
                    {...register("name")}
                    type="text"
                    placeholder="Nombre"
                    id="name"
                    className={`form-input h-10 w-full bg-quaternary p-2 text-sm text-white focus:outline-none ${
                        errors.name ? "border-red-500" : ""
                    }`}
                />
                {errors.name && (
                    <p className="text-xs italic text-primary">
                        {errors.name.message}
                    </p>
                )}
                <input
                    {...register("email")}
                    type="email"
                    placeholder="Email"
                    id="email"
                    className={`form-input h-10 w-full bg-quaternary p-2 text-sm text-white autofill:bg-quaternary focus:outline-none ${
                        errors.email ? "border-red-500" : ""
                    }`}
                />
                {errors.email && (
                    <p className="text-xs italic text-primary">
                        {errors.email.message}
                    </p>
                )}

                <textarea
                    {...register("message")}
                    id="message"
                    placeholder="Mensaje"
                    rows={5}
                    cols={50}
                    className={`form-textarea w-full bg-quaternary p-2 text-sm text-white focus:outline-none ${
                        errors.message ? "border-primary" : ""
                    }`}
                />
                {errors.message && (
                    <p className="text-xs italic text-primary">
                        {errors.message.message}
                    </p>
                )}

                <ReCAPTCHA
                    sitekey={site_key}
                    onChange={onChange}
                    ref={reCaptchaRef}
                    theme="dark"
                />

                {serverError && (
                    <p className="mb-6 text-xs italic text-red-500">
                        {serverError}
                    </p>
                )}

                {success === true && (
                    <p className="mb-6 text-xs italic text-green-500">
                        Message sent successfully!
                    </p>
                )}

                <button
                    type="submit"
                    className={`z-10 inline bg-primary p-4 text-sm font-semibold text-white duration-500 ease-in-out hover:bg-opacity-80
                    ${
                        status === "loading"
                            ? "cursor-not-allowed opacity-50"
                            : ""
                    }`}
                    disabled={status === "loading"}
                >
                    Submit
                </button>
            </form>
        </section>
    );
};

export default Contact;
