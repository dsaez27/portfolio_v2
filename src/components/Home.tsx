import { Link } from "react-scroll";
import { path } from "../helpers/path";

const Home = () => {
    return (
        <>
            <div className="absolute top-0 left-0 h-full w-full">
                <div
                    className="absolute top-0 left-0 h-full w-full bg-cover bg-fixed bg-center"
                    style={{
                        backgroundImage: `url(${path(
                            "background.webp",
                            "images"
                        )})`,
                    }}
                />
                {/*Gradient*/}
                <div className="absolute inset-0 h-full w-full bg-gradient-to-t from-dark via-transparent to-transparent backdrop-blur-sm" />
            </div>

            <section
                id="home"
                className="justify-cente relative flex h-screen w-full flex-col items-start"
            >
                <div className="absolute top-0 left-0 flex w-full flex-row items-center justify-center gap-10">
                    <div className="lg:min-[400px] z-10 flex h-screen w-full flex-col items-start justify-center gap-3 ">
                        <h1 className="z-10 text-5xl font-bold text-white">
                            ¡Hola! Soy <br />
                            Daniel Sáez H.
                        </h1>
                        <h5 className="z-10 pb-4 text-sm text-white">
                            "Soy un desarrollador web con experiencia en React,
                            Redux, CSS3, Sass, Bootstrap y Tailwind. Manejo Git
                            y Github para colaboración en equipo. En mi sitio
                            web encontrarás mis proyectos personales y
                            habilidades de programación. Contáctame si necesitas
                            ayuda en el desarrollo de un proyecto. ¡Gracias por
                            visitar mi sitio!
                        </h5>
                        <div className="flex flex-row gap-2 py-3">
                            <a
                                href="https://www.linkedin.com/in/dsaezh/"
                                target="_blank"
                                rel="noreferrer"
                                data-title="Linkedin"
                            >
                                <svg
                                    xmlns="http://www.w3.org/2000/svg"
                                    width="36"
                                    height="36"
                                    viewBox="0 0 24 24"
                                    fill="#fff"
                                    className="rounded-full border-2 p-1 transition duration-300  ease-in-out hover:border-primary hover:fill-primary"
                                >
                                    <path d="M12 0c-6.627 0-12 5.373-12 12s5.373 12 12 12 12-5.373 12-12-5.373-12-12-12zm-2 16h-2v-6h2v6zm-1-6.891c-.607 0-1.1-.496-1.1-1.109 0-.612.492-1.109 1.1-1.109s1.1.497 1.1 1.109c0 .613-.493 1.109-1.1 1.109zm8 6.891h-1.998v-2.861c0-1.881-2.002-1.722-2.002 0v2.861h-2v-6h2v1.093c.872-1.616 4-1.736 4 1.548v3.359z" />
                                </svg>
                            </a>
                            <a
                                href="https://github.com/dsaez27"
                                target="_blank"
                                rel="noreferrer"
                                data-title="Github"
                            >
                                <svg
                                    xmlns="http://www.w3.org/2000/svg"
                                    width="36"
                                    height="36"
                                    viewBox="0 0 24 24"
                                    fill="#fff"
                                    className="rounded-full border-2 p-1 transition duration-300  ease-in-out hover:border-primary hover:fill-primary"
                                >
                                    <path d="M12 0c-6.626 0-12 5.373-12 12 0 5.302 3.438 9.8 8.207 11.387.599.111.793-.261.793-.577v-2.234c-3.338.726-4.033-1.416-4.033-1.416-.546-1.387-1.333-1.756-1.333-1.756-1.089-.745.083-.729.083-.729 1.205.084 1.839 1.237 1.839 1.237 1.07 1.834 2.807 1.304 3.492.997.107-.775.418-1.305.762-1.604-2.665-.305-5.467-1.334-5.467-5.931 0-1.311.469-2.381 1.236-3.221-.124-.303-.535-1.524.117-3.176 0 0 1.008-.322 3.301 1.23.957-.266 1.983-.399 3.003-.404 1.02.005 2.047.138 3.006.404 2.291-1.552 3.297-1.23 3.297-1.23.653 1.653.242 2.874.118 3.176.77.84 1.235 1.911 1.235 3.221 0 4.609-2.807 5.624-5.479 5.921.43.372.823 1.102.823 2.222v3.293c0 .319.192.694.801.576 4.765-1.589 8.199-6.086 8.199-11.386 0-6.627-5.373-12-12-12z" />
                                </svg>
                            </a>
                            <a
                                href="mailto:daniel.s.saez.h@gmail.com"
                                target="_blank"
                                rel="noreferrer"
                                data-title="Email"
                            >
                                <svg
                                    xmlns="http://www.w3.org/2000/svg"
                                    width="36"
                                    height="36"
                                    viewBox="0 0 24 24"
                                    fill="#fff"
                                    className="rounded-full border-2 p-1 transition duration-300  ease-in-out hover:border-primary hover:fill-primary"
                                >
                                    <path d="M12 .02c-6.627 0-12 5.373-12 12s5.373 12 12 12 12-5.373 12-12-5.373-12-12-12zm6.99 6.98l-6.99 5.666-6.991-5.666h13.981zm.01 10h-14v-8.505l7 5.673 7-5.672v8.504z" />
                                </svg>
                            </a>
                            <a
                                href={path("cv.pdf", "doc")}
                                data-title="Descargar CV"
                                download
                            >
                                <svg
                                    width="36px"
                                    height="36px"
                                    viewBox="0 0 24 24"
                                    fill="none"
                                    xmlns="http://www.w3.org/2000/svg"
                                    className="rounded-full border-2 p-1 transition duration-300  ease-in-out hover:border-primary hover:fill-primary"
                                >
                                    <path
                                        opacity="0.15"
                                        d="M5 21H19V9H15C13.8954 9 13 8.10457 13 7V3H5V21Z"
                                        fill="#fff"
                                    />
                                    <path
                                        d="M9 14H15M13 3H5V21H19V9M13 3H14L19 8V9M13 3V7C13 8 14 9 15 9H19"
                                        stroke="#fff"
                                        strokeLinecap="round"
                                        strokeLinejoin="round"
                                    />
                                </svg>
                            </a>
                        </div>
                    </div>
                    <img
                        src={path("astronot.webp", "images")}
                        alt="astronaut"
                        className="w-[300px] animate-astronaut max-sm:hidden"
                        style={{
                            transform: "scaleX(-1)",
                        }}
                    />
                </div>
            </section>
        </>
    );
};

export default Home;
