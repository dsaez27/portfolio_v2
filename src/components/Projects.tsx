import { projects } from "@/data";
import { path } from "@/helpers";

const Projects = () => {
    return (
        <section id="cards" className="h-full w-full">
            <h2 className="text-3xl font-semibold text-primary after:mt-1 after:block after:h-1 after:w-20 after:bg-primary">
                Proyectos
            </h2>
            <div className="cards__grid my-10 grid justify-items-center gap-3">
                {projects.map(({ id, title, desc, img }) => (
                    <figure
                        className="card relative h-full w-full cursor-pointer bg-cover bg-center bg-no-repeat text-left text-base text-white"
                        key={id}
                        style={{
                            backgroundImage: `url(${path(img, "images")})`,
                        }}
                    >
                        <div className="card__content absolute inset-0 flex flex-col justify-end bg-opacity-25 bg-card p-2">
                            <h3 className="card__heading text-lg">{title}</h3>
                            <p className="card__desc text-sm">{desc}</p>
                            <div className="card__buttons absolute inset-0 flex flex-col items-center justify-center gap-1 opacity-0">
                                <button className="mb-2 w-24 rounded-full bg-primary p-1 text-sm text-white transition duration-500 ease-in-out hover:bg-secondary">
                                    Ver CÃ³digo
                                </button>
                                <button className="mb-2 w-24 rounded-full bg-primary p-1 text-sm text-white transition duration-500 ease-in-out hover:bg-secondary">
                                    Demo
                                </button>
                            </div>
                        </div>
                    </figure>
                ))}
            </div>
        </section>
    );
};

export default Projects;
