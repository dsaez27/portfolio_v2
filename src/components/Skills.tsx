import { path } from "@/helpers";
import { skills } from "../data";

const Skills = () => {
    return (
        <section id="skills" className="w-full h-full">
            <h2 className="after-content-[''] text-3xl font-semibold text-primary after:mt-1  after:block after:h-1 after:w-20 after:bg-primary">
                Skills
            </h2>
            <div
                className="my-10 grid justify-items-center gap-3"
                style={{
                    gridTemplateColumns: "repeat(auto-fit, minmax(135px, 1fr))",
                    gridAutoRows: "minmax(100px, auto)",
                }}
            >
                {skills.map((skill) => {
                    return (
                        <figure
                            key={skill.id}
                            className="group flex w-full flex-col items-center justify-center bg-tertiary align-middle"
                        >
                            <svg
                                xmlns="http://www.w3.org/2000/svg"
                                width="48"
                                height="48"
                                fill="#fff"
                                className="p-1 transition duration-300  ease-in-out hover:fill-primary group-hover:fill-primary"
                                viewBox={skill.viewBox}
                                id={skill.name}
                            >
                                <path d={skill.icon} />
                            </svg>
                            <figcaption className="mt-2 text-xs text-white transition duration-300  ease-in-out group-hover:text-primary cursor-default">
                                {skill.name}
                            </figcaption>
                        </figure>
                    );
                })}
            </div>
        </section>
    );
};

export default Skills;
