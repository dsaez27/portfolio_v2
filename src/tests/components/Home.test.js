
import { Home } from './../../components/Home';
import { shallow } from 'enzyme';

describe('Pruebas en <Home />', () => {
    test('Debe de mostrarse correctamente', () => {
        const wrapper = shallow(<Home />);
        expect(wrapper).toMatchSnapshot();
    })
    
})
