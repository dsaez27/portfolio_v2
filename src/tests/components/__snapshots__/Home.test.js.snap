// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Home /> Debe de mostrarse correctamente 1`] = `
<Fragment>
  <div
    id="home"
  />
  <v
    height="100%"
    id="tsparticles"
    options={
      Object {
        "interactivity": Object {
          "detect_on": "canvas",
          "events": Object {
            "onclick": Object {
              "enable": false,
              "mode": "repulse",
            },
            "onhover": Object {
              "enable": false,
              "mode": "bubble",
            },
            "resize": true,
          },
          "modes": Object {
            "bubble": Object {
              "distance": 250,
              "duration": 2,
              "opacity": 0,
              "size": 0,
              "speed": 3,
            },
            "grab": Object {
              "distance": 0,
              "line_linked": Object {
                "opacity": 1,
              },
            },
            "push": Object {
              "particles_nb": 4,
            },
            "remove": Object {
              "particles_nb": 2,
            },
            "repulse": Object {
              "distance": 400,
              "duration": 0.4,
            },
          },
        },
        "particles": Object {
          "color": Object {
            "value": "#ffffff",
          },
          "line_linked": Object {
            "color": "#ffffff",
            "distance": 150,
            "enable": false,
            "opacity": 0.4,
            "width": 1,
          },
          "move": Object {
            "attract": Object {
              "enable": false,
              "rotateX": 600,
              "rotateY": 600,
            },
            "bounce": false,
            "direction": "none",
            "enable": true,
            "out_mode": "out",
            "random": true,
            "speed": 1,
            "straight": false,
          },
          "number": Object {
            "density": Object {
              "enable": true,
              "value_area": 1893.9543399174545,
            },
            "value": 246,
          },
          "opacity": Object {
            "anim": Object {
              "enable": true,
              "opacity_min": 0,
              "speed": 1,
              "sync": false,
            },
            "random": true,
            "value": 1,
          },
          "shape": Object {
            "image": Object {
              "height": 100,
              "src": "img/github.webp",
              "width": 100,
            },
            "polygon": Object {
              "nb_sides": 5,
            },
            "stroke": Object {
              "color": "#000000",
              "width": 0,
            },
            "type": "circle",
          },
          "size": Object {
            "anim": Object {
              "enable": false,
              "size_min": 1,
              "speed": 10,
              "sync": false,
            },
            "random": true,
            "value": 1,
          },
        },
        "retina_detect": true,
      }
    }
    style={
      Object {
        "padding": "0",
      }
    }
    width="100%"
  />
  <Draw
    style={
      Object {
        "zIndex": "1",
      }
    }
  />
  <styled.div
    style={
      Object {
        "zIndex": "2",
      }
    }
  >
    <styled.h1>
      Hello I'm a 
      <br />
      Front-end Developer
    </styled.h1>
    <styled.p>
      Bienvenido a mi portafolio personal, aquí podrás ver algunos de mis proyectos, con una gran cantidad de componentes, estilos y tecnologías diferentes.
    </styled.p>
    <button>
      Ver Proyectos
    </button>
  </styled.div>
</Fragment>
`;
